src/ruuvitag_ble/parser.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import logging
+
+from bluetooth_sensor_state_data import BluetoothData
+from home_assistant_bluetooth import BluetoothServiceInfo
+from sensor_state_data import DeviceClass, Units
+
+from ruuvitag_ble.df5_decoder import DataFormat5Decoder
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RuuvitagBluetoothDeviceData(BluetoothData):
+    """Data for Ruuvitag BLE sensors."""
+
+    def _start_update(self, service_info: BluetoothServiceInfo) -> None:
+        try:
+            raw_data = service_info.manufacturer_data[0x0499]
+        except (KeyError, IndexError):
+            _LOGGER.debug("Manufacturer ID 0x0499 not found in data")
+            return None
+
+        data_format = raw_data[0]
+        if data_format != 0x05:
+            _LOGGER.debug("Data format not supported: %s", raw_data)
+            return
+
+        decoder = DataFormat5Decoder(raw_data)
+        mac = str(decoder.mac or service_info.address)  # Prefer MAC from advertisement.
+        self.set_device_type("RuuviTag")
+        self.set_device_manufacturer("Ruuvi Innovations Ltd.")
+        self.set_device_name(f"RuuviTag {mac}")
+
+        self.update_sensor(
+            key=DeviceClass.TEMPERATURE,
+            device_class=DeviceClass.TEMPERATURE,
+            native_unit_of_measurement=Units.TEMP_CELSIUS,
+            native_value=decoder.temperature_celsius,
+        )
+        self.update_sensor(
+            key=DeviceClass.HUMIDITY,
+            device_class=DeviceClass.HUMIDITY,
+            native_unit_of_measurement=Units.PERCENTAGE,
+            native_value=decoder.humidity_percentage,
+        )
+        self.update_sensor(
+            key=DeviceClass.PRESSURE,
+            device_class=DeviceClass.PRESSURE,
+            native_unit_of_measurement=Units.PRESSURE_HPA,
+            native_value=decoder.pressure_hpa,
+        )
+        self.update_sensor(
+            key=DeviceClass.VOLTAGE,
+            device_class=DeviceClass.VOLTAGE,
+            native_unit_of_measurement=Units.ELECTRIC_POTENTIAL_MILLIVOLT,
+            native_value=decoder.battery_voltage_mv,
+        )
+        self.update_sensor(
```

Comment:
I think you can also add the binary sensor "motion". I know Home Assistant requires you to first implement only one platform (e.g. sensors), and later add different  ones (e.g. binary sensors). But you can already add it.

Reply:
I'm not sure I follow how a movement counter would translate into a binary sensor. I'd have to keep track of what the movement counter had been, and that sounds terribly stateful...?

Reply:
In BLE monitor, I use the following logic. 

If the movement counter is increased between two advertisements, the motion sensor becomes True (in HA True = motion detected). 
After a certain time, without a movement counter change, the motion sensor returns to False (no motion detected). This probably has to be done in the HA integration code. 

In that way, you can use it as a motion detector. 

I'm not saying you have to implement this, but people might complain later that there isn't a motion binary sensor. But we can always add this later. Let's first implement the basics, which is the movement counter (as that is what the sensor broadcasts).